libfwupdplugin/fu-firmware.c

```diff
@@ -707,6 +707,34 @@ fu_firmware_get_stream(FuFirmware *self, GError **error)
 	return NULL;
 }
 
+/**
+ * fu_firmware_set_stream:
+ * @self: a #FuPlugin
+ * @stream: (nullable): #GInputStream
+ * @error: (nullable): optional return location for an error
+ *
+ * Sets the input stream.
+ *
+ * Returns: %TRUE on success
+ *
+ * Since: 2.0.0
+ **/
+gboolean
+fu_firmware_set_stream(FuFirmware *self, GInputStream *stream, GError **error)
+{
+	FuFirmwarePrivate *priv = GET_PRIVATE(self);
+	g_return_val_if_fail(FU_IS_FIRMWARE(self), FALSE);
+	g_return_val_if_fail(stream == NULL || G_IS_INPUT_STREAM(stream), FALSE);
+	if (stream != NULL) {
+		if (!fu_input_stream_size(stream, &priv->streamsz, error))
+			return FALSE;
+	} else {
```

Comment:
This seems impossible with the g return values above

Reply:
I don't think I agree, it's:

    	g_return_val_if_fail(stream == NULL || G_IS_INPUT_STREAM(stream), FALSE);

i.e. it can be a valid GInputStream **or** NULL.

Reply:
Ah missed that second part. Ok.